### PR TITLE
Added Liberica Native Image Kit as new distribution

### PR DESCRIPTION
**Pull request by petermz**
This PR addresses #76. Liberica is supported in pretty much the same fashion as GraalVM, e.g.
```
distribution: liberica
java-version: 17.0.10
```
It appeared logical to me to reuse several functions from `utils.ts`, so I converted a few constants used there into parameters. Also added tests, notes, and bumped version.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -243,6 +243,48 @@ jobs:
...
+        version: ['', 'std', 'full']
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run setup-graalvm action
+        uses: ./
+        with:
+          distribution: liberica
+          java-version: ${{ matrix.java-version }}
+          version: ${{ matrix.version }}
```
Comment (fniephaus) [inline .github/workflows/test.yml:262]: We actually am phasing out `version` in favor of just the `java-version`. It seems there are different variants of Liberica. Can these also be pulled by `setup-java` and if yes, how?

**Review by fniephaus — COMMENTED** [review]
Thanks for the PR, I'll try to find some time to review it in the next couple of days.

File: .github/workflows/test.yml (hunk 1 above)
Comment (petermz) [reply .github/workflows/test.yml:262]: AFAICT this is controlled by the `java-package` parameter:

> java-package: The packaging variant of the chosen distribution. Possible values: jdk, jre, jdk+fx, jre+fx. Default value: jdk

The value of `jdk` corresponds to `version: std`, and `jdk+fx` corresponds to `version: full`

Comment (fniephaus) [reply .github/workflows/test.yml:262]: Thanks! Mapping `jdk+fx` to `full` seems like a hack, assuming fx stands for JavaFX? If Liberica users are more familiar with `std` and `full`, I'm ok with using those directly, maybe also allowing `jdk` and `jdk+fx` for full compatibility with `setup-java`. In any case, we should add [the `java-package` input from `setup-java`](https://github.com/actions/setup-java/blob/9704b39bf258b59bc04b50fa2dd55e9ed76b47a8/action.yml#L13-L16) and not use the `version` input for this.

Comment (petermz) [reply .github/workflows/test.yml]: Ok, I've added the `java-package: jdk | jdk+fx` input

**Comment by TKaxv-7S** [comment]
Hi, I was Fixed 'version' parameter at this [Pull](https://github.com/petermz/setup-graalvm/pull/1)

File: src/liberica.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    return `linux-${c.JDK_ARCH}-musl`
+  } else {
+    return `${c.JDK_PLATFORM}-${c.GRAALVM_ARCH}`
+  }
+}
+
+function isMuslBasedLinux() {
+  if (c.IS_LINUX) {
+    const output = spawnSync('ldd', ['--version']).stderr.toString('utf8')
+    if (output.indexOf('musl') > -1) {
```
Comment (fniephaus) [inline src/liberica.ts:110]:
```suggestion
    if (output.includes('musl')) {
```

File: src/liberica.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  if (isMuslBasedLinux()) {
+    return `linux-${c.JDK_ARCH}-musl`
+  } else {
+    return `${c.JDK_PLATFORM}-${c.GRAALVM_ARCH}`
+  }
+}
+
+function isMuslBasedLinux() {
+  if (c.IS_LINUX) {
+    const output = spawnSync('ldd', ['--version']).stderr.toString('utf8')
```
Comment (fniephaus) [inline src/liberica.ts:109]: So you'll automatically switch to a musl version of Liberica on musl-based Linux system. Are you sure GitHub Action can provide musl-based systems? If yes, how, via Docker? Even though `setup-java` [mentions `linux-musl`](https://github.com/actions/setup-java/blob/99b8673ff64fbf99d8d325f52d9a5bdedb8483e9/src/distributions/liberica/installer.ts#L19), it does not seems to be [actually supported](https://github.com/actions/setup-java/blob/99b8673ff64fbf99d8d325f52d9a5bdedb8483e9/src/distributions/liberica/installer.ts#L159-L172).

Maybe this additional logic can be safely removed to keep things simple?

**Review by fniephaus — COMMENTED** [review]
I went through the changes and they look good. I left a couple of comments. If you could address them, that'd be great. Thanks!

**Comment by fniephaus** [comment]
The [failure on Windows](https://github.com/graalvm/setup-graalvm/actions/runs/8322999223/job/22774508807?pr=87) is unrelated to this PR. I'll take care of it.

**Comment by TKaxv-7S** [comment]
> Hi, I was Fixed 'version' parameter at this [Pull](https://github.com/petermz/setup-graalvm/pull/1)

This PULL also fixed the issue of not being able to download the latest version of graalvm.

File: src/liberica.ts (hunk 3 above)
Comment (petermz) [reply src/liberica.ts:109]: I believe they use Docker images, and do have at least [some support](https://github.com/actions/runner-images/pull/7274) for Alpine. Plus, people can set up their own self-hosted runners.

**Comment by petermz** [comment]
Addressed rest of the comments

File: src/liberica.ts (hunk 3 above)
Comment (fniephaus) [reply src/liberica.ts:109]: Just wanted to point out this is feature isn't really tested anywhere.

Comment (petermz) [reply src/liberica.ts:109]: True; I've done some manual testing inside a container, but I'm not sure how to automate that

**Comment by fniephaus** [comment]
This is now live and available via `v1.2.0`. I rebase the commits and unfortunately GitHub didn't detect that, but this PR is merged now. Again, thanks for the contribution!

**Pull request closed without merging**